PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/mysql/ServerNameTextField.java

```diff
@@ -0,0 +1,98 @@
+ * SOFTWARE.
+ */
+package com.microsoft.azure.toolkit.intellij.mysql;
+
+import com.microsoft.azure.management.mysql.v2020_01_01.Server;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.Subscription;
+import com.microsoft.azure.toolkit.intellij.common.ValidationDebouncedTextInput;
+import com.microsoft.azure.toolkit.lib.common.form.AzureValidationInfo;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.core.mvp.model.mysql.MySQLMvpModel;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+public class ServerNameTextField extends ValidationDebouncedTextInput {
+    private Subscription subscription;
+    private ResourceGroup resourceGroup;
+
+    private final Pattern pattern = Pattern.compile("^[a-z0-9][a-z0-9-]+[a-z0-9]$");
+
+    public void setSubscription(Subscription subscription) {
+        if (!Objects.equals(subscription, this.subscription)) {
+            this.subscription = subscription;
+        }
+    }
+
+    public void setResourceGroup(ResourceGroup resourceGroup) {
+        if (!Objects.equals(resourceGroup, this.resourceGroup)) {
+            this.resourceGroup = resourceGroup;
+            this.revalidateValue();
+        }
+    }
+
+    /**
+     * Server name must be at least 3 characters and at most 63 characters.
+     * Server name must only contain lowercase letters, numbers, and hyphens. The server name must not start or end in a hyphen.
+     * Server name must be available.
+     */
+    @NotNull
+    public AzureValidationInfo doValidateValue() {
+        final AzureValidationInfo info = super.doValidateValue();
+        if (!AzureValidationInfo.OK.equals(info)) {
+            return info;
+        }
+        final String value = this.getValue();
+        // validate length
+        if (StringUtils.length(value) < 3 || StringUtils.length(value) > 63) {
+            final AzureValidationInfo.AzureValidationInfoBuilder builder = AzureValidationInfo.builder();
+            return builder.input(this).message("Server name must be at least 3 characters and at most 63 characters.")
+                    .type(AzureValidationInfo.Type.ERROR).build();
+        }
+        // validate special character
+        if (!pattern.matcher(value).matches()) {
+            final AzureValidationInfo.AzureValidationInfoBuilder builder = AzureValidationInfo.builder();
+            return builder.input(this)
+                    .message("Server name must only contain lowercase letters, numbers, and hyphens. The server name must not start or end in a hyphen.")
+                    .type(AzureValidationInfo.Type.ERROR).build();
+        }
+        // validate availability
+        try {
+            Server server = MySQLMvpModel.findServer(subscription.subscriptionId(), resourceGroup.name(), this.getValue());
```

Comment:
the name confliction check is wrong, you need to special check name availability api.

Reply:
you should check the name outside this subs for global exclusion.

Reply:
https://github.com/Azure/azure-libraries-for-java/blob/master/azure-mgmt-appservice/src/main/java/com/microsoft/azure/management/appservice/CheckNameResourceTypes.java

Reply:
![image](https://user-images.githubusercontent.com/17075863/102059948-883b4180-3e2c-11eb-8a54-9ad9c59b108f.png)

Reply:
No API for MySQL. if a server name does not exist, SDK find method will return NPE.